mypy/test/test_check.py

```diff
@@ -122,13 +122,12 @@ def run_test_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
             a = e.messages
         a = normalize_error_messages(a)
 
-        if output != a and mypy.myunit.UPDATE_TESTCASES:
+        if output != a and pytest.config.getoption('UPDATE_TESTCASES'):
             update_testcase_output(testcase, a, mypy.myunit.APPEND_TESTCASES)
 
-        assert_string_arrays_equal(
-            output, a,
+        assert output == a, \
             'Invalid type checker output ({}, line {})'.format(
-                testcase.file, testcase.line))
+                testcase.file, testcase.line)
```

Comment:
@gnprice @gvanrossum If I change the assertion to `assert '\n'.join(output) == '\n'.join(a)`, the diff is actually really nice:

```
============================================ FAILURES ============================================
___________________________ TestTypeCheck.testListAssignmentFromTuple ____________________________

self = <mypy.test.test_check.TestTypeCheck object at 0x7f7145179588>
test = <mypy.test.data.DataDrivenTestCase object at 0x7f714557d780>

    def func(self, test):
>       test.run(self)

../../mypy/test/helpers.py:314: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../mypy/test/data.py:163: in run
    self.perform(obj, self)
../../mypy/test/test_check.py:82: in run_test
    self.run_test_once(testcase)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <mypy.test.test_check.TestTypeCheck object at 0x7f7145179588>
testcase = <mypy.test.data.DataDrivenTestCase object at 0x7f714557d780>, incremental = 0

    def run_test_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
        find_module_clear_caches()
        pyversion = pyversion_testcase(testcase.file, testcase.name)
        program_text = '\n'.join(testcase.input)
        module_name, program_name, program_text = self.parse_options(program_text)
        flags = self.parse_flags(program_text)
        output = testcase.output
        if incremental:
            flags.append(build.INCREMENTAL)
            if incremental == 1:
                # In run 1, copy program text to program file.
                output = []
                with open(program_name, 'w') as f:
                    f.write(program_text)
                    program_text = None
            elif incremental == 2:
                # In run 2, copy *.py.next files to *.py files.
                for dn, dirs, files in os.walk(os.curdir):
                    for file in files:
                        if file.endswith('.py.next'):
                            full = os.path.join(dn, file)
                            target = full[:-5]
                            shutil.copy(full, target)
        source = BuildSource(program_name, module_name, program_text)
        try:
            res = build.build(target=build.TYPE_CHECK,
                              sources=[source],
                              pyversion=pyversion,
                              flags=flags + [build.TEST_BUILTINS],
                              alt_lib_path=test_temp_dir)
            a = res.errors
        except CompileError as e:
            res = None
            a = e.messages
        a = normalize_error_messages(a)
    
        if output != a and pytest.config.getoption('UPDATE_TESTCASES'):
            update_testcase_output(testcase, a, mypy.myunit.APPEND_TESTCASES)
    
>       assert '\n'.join(output) == '\n'.join(a), \
            'Invalid type checker output ({}, line {})'.format(
                testcase.file, testcase.line)
E       AssertionError: Invalid type checker output (mypy/test/data/check-lists.test, line 34)
E       assert 'main:6: erro..., 2 provided)' == 'main:6: error..., 2 provided)'
E         Skipping 137 identical leading characters in diff, use -v to show
E         Skipping 67 identical trailing characters in diff, use -v to show
E         - o unpack (4 expected
E         ?           ^
E         + o unpack (3 expected
E         ?           ^

../../mypy/test/test_check.py:128: AssertionError
====================== 1860 tests deselected by '-kListAssignmentFromTuple' ======================
```

Even if `assert_string_arrays_equal` is used, the function body will be dumped; this is more of a side-effect of pytest than of the assertion.